include/sys/zfs_context.h

```diff
@@ -273,6 +273,7 @@ typedef struct kmutex {
 } kmutex_t;
 
 #define	MUTEX_DEFAULT	0
+#define	MUTEX_FSTRANS	MUTEX_DEFAULT
```

Comment:
@yarikoptic the key bit of information you need to be aware of is that large parts of ZoL are compiled twice when you build it.  The first time it build the core ZFS code in user space for ztest and the user space utilities.  The second time it builds it as kernel modules.  This means we have two mutex implementations one for user space and one for the kernel.

This commit only ensures that this flag is understood in user space so everything builds.  As you correctly pointed out it doesn't actually fix anything, but that's fine because in user space this deadlock can't occur.  It's the counter part SPL patches where the fix is, that's the kernel mutex implementation.  Those patches were updated so they disable part of the kernel's direct reclaim path while the mutex is held.  This prevents the deadlock from being possible.